### DISCUSSION TITLE
Taking Build #2583 to Worlds

### DISCUSSION THREAD

**Discussion by TheWendyPower** [pybricks/support#1053] [Show and tell]
I was going to add part of this to [#989](https://github.com/pybricks/support/issues/989). 

We knew that we were using beta firmware, and I had a talk with the kids about what that meant before they voted to keep using it. We also didn't catch this until we were pretty much done and had no time to revert. When turning the robot on and doing the initial runs there would be errors, as in it wouldn't go where we expected it to go. Then rerunning it, the robot would move just as planned, completing the tasks. At first, I chalked this up to the kids not positioning the robot in the right place. However, time and time again the first run, after turning on the robot, didn't work correctly, and subsequently runs went as planned. 

On our 3rd official run at Worlds, we tried to test this. One of the students held the robot as they ran through all of their robot runs. Unfortunately, the robot went to sleep before we were able to get it to the game table. We have no regrets over the version we took with us. It was a great learning experience. Overall the team has learned a lot about robot design and code over this season. We look forward to taking those lessons and making a more robust robot for next year. 

I know I can't wait to play with the new version that just dropped on beta.pybricks.com

The team brought home a "Rising Star" award:
![Rising_Star](https://user-images.githubusercontent.com/43281569/235160681-74d533f7-f453-4761-b826-25710648cc6a.jpg)

And got to work with a fantastic team from Australia (Donut Eating Llamas) as our alliance partner in the Encore game: 
![OnePlus_20230422_111042](https://user-images.githubusercontent.com/43281569/235161893-c5f647c9-b284-4935-9639-deb9ec129636.jpg)

**Comment by laurensvalk** [comment]
Thanks so much for sharing your team's story. Congrats to everyone for making it all the way to worlds and winning the Rising Star award!

> I know I can't wait to play with the new version that just dropped on beta.pybricks.com

Thank you. I'm sorry about the issues you ran into with the experimental firmware. I hope this will not discourage the team when taking part next year.

If you see the same with the firmware in the official beta channel, please do let us know. I think the issue you describe has been fixed in the mean time, but otherwise we'd certainly aim to fix it for the stable release.

**Reply by TheWendyPower** [reply]
They are definitely not discouraged. I stressed to the kids, if they choose to use experimental firmware they also accept that there are going to be bugs. It is experimental for a reason. The plan is to go into next season using Pybricks from the start.

**Reply by laurensvalk** [reply]
I'm glad to hear it. :slightly_smiling_face: 

If it helps for closure, I think this particular bug could happen if the `DriveBase` was initialized **before** the `hub` was initialized. Upon initialization, the `DriveBase` takes one heading measurement as its starting point, but that would be undefined in that case.

By the second time you ran your program, the hub was initialized at least once, so it worked as intended.

This has been fixed now. The `GyroDriveBase` can be used now even if the `hub` is not instantiated at all.
